### ISSUE TITLE
Advanced search for application by Bundle Name fails

### ISSUE THREAD

**Issue by jokdbx** [salopensource/sal#279]
```[15/Aug/2018 19:41:15] ERROR [django.request:124] Internal Server Error: /search/run_search/88/
Traceback (most recent call last):
  File "/usr/local/sal_install-3.3.8/sal_env/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/sal_install-3.3.8/sal_env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/sal_install-3.3.8/sal_env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/sal_install-3.3.8/sal_env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/sal_install-3.3.8/sal_env/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/usr/local/sal_install-3.3.8/sal/search/views.py", line 289, in run_search
    machines = search_machines(search_id, machines)
  File "/usr/local/sal_install-3.3.8/sal/search/views.py", line 181, in search_machines
    'inventoryitem__application__%s%s' % (search_field, operator): search_row.search_term  # noqa: E501
UnboundLocalError: local variable 'search_field' referenced before assignment
```

I believe only 2 of the 4 options are handled in this view.

**Comment by grahamgilbert** [comment]
A screenshot of your exact query would help here.

**Comment by jokdbx** [comment]
the boxes on my search contains:
```
Search Item | Application Inventory
Bundle Name
Contains
munki

```

The code line 175 through 179 only handles 'name' or 'bundleid'. neither 'path' nor 'bundlename' are handled as a search, but are offered in the dropdown in advanced search.

**Comment by grahamgilbert** [comment]
Then as you’ve found the issue, you can PR it to fix it ;)

**Comment by grahamgilbert** [comment]
Fixed in #282
